integration-tests/jpa-postgresql-vault/src/main/java/io/quarkus/it/jpa/postgresql/JPAFunctionalityTestEndpoint.java

```diff
@@ -0,0 +1,167 @@
+package io.quarkus.it.jpa.postgresql;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.EntityTransaction;
+import javax.persistence.PersistenceUnit;
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Various tests covering JPA functionality. All tests should work in both standard JVM and SubstrateVM.
+ */
+@WebServlet(name = "JPATestBootstrapEndpoint", urlPatterns = "/jpa/testfunctionality")
+public class JPAFunctionalityTestEndpoint extends HttpServlet {
```

Comment:
I think we would need to make the tests more specific to Vault. Here we have copied some complicated model that we don't need and all the names and comments are related to testing JPA functionalities.

Reply:
done, as discussed in https://github.com/quarkusio/quarkus/pull/2897#discussion_r329320973